src/main/kotlin/FilePathGenerator.kt

```diff
@@ -0,0 +1,103 @@
+        private fun getCompressedTypes(): Array<SupportedExtensions> = arrayOf(MP3, JPEG, JPG)
+
+        fun isSupportedExtension(ext: String): Boolean = values().any { it.name == ext.toUpperCase() }
+
+        fun isSupportedContainer(ext: String): Boolean = getSupportedContainers().contains(valueOf(ext.toUpperCase()))
+        fun isCompressedType(ext: String): Boolean = getCompressedTypes().contains(valueOf(ext.toUpperCase()))
+    }
+}
+
+enum class Groupings(val grouping: String) {
+    BOOK("book"),
+    CHAPTER("chapter"),
+    VERSE("verse"),
+    CHUNK("chunk");
+
+    companion object {
+        fun isSupportedGrouping(grouping: String): Boolean = values().any { it.name == grouping.toUpperCase() }
+    }
+}
+
+enum class MediaQuality(val quality: String) {
+    HI("hi"),
+    LOW("low");
+
+    companion object {
+        fun isSupportedQuality(quality: String): Boolean = values().any { it.name == quality.toUpperCase() }
+    }
+}
+
+object FilePathGenerator {
+
+    fun createPathFromFile(
+        inputFile: File,
+        languageCode: String,
+        dublinCoreId: String,
+        grouping: String,
+        projectId: String = "",
+        mediaExtension: String = "",
+        mediaQuality: String = ""
+    ): String {
+
+        validateInput(inputFile, languageCode, dublinCoreId, grouping, mediaExtension, mediaQuality)
+
+        val projectPath = if (projectId.isBlank()) "" else "$projectId/"
+        val compressionQuality = if (mediaQuality.isBlank()) "hi" else mediaQuality
+
+        var path = "$languageCode/$dublinCoreId/${projectPath}CONTENTS/${inputFile.extension}/"
+
+        if(SupportedExtensions.isSupportedContainer(inputFile.extension)) {
+            path += if (SupportedExtensions.isCompressedType(mediaExtension)) "$mediaExtension/$compressionQuality/" else "$mediaExtension/"
+        } else {
+            if (SupportedExtensions.isCompressedType(inputFile.extension)) path += "$compressionQuality/"
+        }
+
+        path += "$grouping/${inputFile.name}"
+
+        return path
+
+    }
+
+    @Throws(IllegalArgumentException::class)
+    fun validateInput(
+        inputFile: File,
+        languageCode: String,
+        dublinCoreId: String,
+        grouping: String,
+        mediaExtension: String = "",
+        mediaQuality: String = ""
+    ) {
+
+        if (languageCode.isBlank()) throw IllegalArgumentException("Language Code is empty")
+        if (dublinCoreId.isBlank()) throw IllegalArgumentException("Dublin Core ID is empty")
+
+        if (grouping.isBlank()) throw IllegalArgumentException("Group is empty")
+        if (!Groupings.isSupportedGrouping(grouping)) throw IllegalArgumentException("Group is not supported")
+
+        validateExtensions(inputFile.extension, mediaExtension)
+
+        if (!MediaQuality.isSupportedQuality(mediaQuality) && mediaQuality.isNotBlank()) throw IllegalArgumentException("Media Quality is invalid")
```

Comment:
why not put this next to line 87?

Reply:
Why not make your isSupportedQuality sufficient enough to determine if the quality is supported? Either it is supported or it isn't? You shouldn't need an extra call here to validate that.